pkg/specs/chart.go

```diff
@@ -295,3 +296,97 @@ func (r *Resolver) calculateContentSHA(root string) (string, error) {
 
 	return fmt.Sprintf("%x", sha256.Sum256(contents)), nil
 }
+
+// this function is not perfect, and has known limitations. One of these is that it does not account for `\n---\n` in multiline strings.
+func (r *Resolver) maybeSplitMultidocYaml(ctx context.Context, localPath string) error {
+	type outputYaml struct {
+		name     string
+		contents string
+	}
+
+	type minimalK8sYaml struct {
+		Kind     string `json:"kind" yaml:"kind" hcl:"kind"`
+		Metadata struct {
+			Name      string `json:"name" yaml:"name" hcl:"name"`
+			Namespace string `json:"namespace" yaml:"namespace" hcl:"namespace"`
+		}
+	}
+
+	files, err := r.FS.ReadDir(localPath)
+	if err != nil {
+		return errors.Wrapf(err, "read files in %s", localPath)
+	}
+
+	if len(files) != 1 {
+		// if there's more than one file, we'll assume that it does not need to be split
+		// if there are no files, there's nothing to do
+		return nil
+	}
+
+	file := files[0]
+
+	if file.IsDir() {
+		// if the single file is a directory, obviously we can't split it
+		return nil
+	}
+
+	if filepath.Ext(file.Name()) != ".yaml" && filepath.Ext(file.Name()) != ".yml" {
+		// not yaml, nothing to do
+		return nil
+	}
+
+	inFileBytes, err := r.FS.ReadFile(filepath.Join(localPath, file.Name()))
+	if err != nil {
+		return errors.Wrapf(err, "read %s", filepath.Join(localPath, file.Name()))
+	}
+
+	outputFiles := []outputYaml{}
+	filesStrings := strings.Split(string(inFileBytes), "\n---\n")
+
+	if len(filesStrings) == 1 {
+		// not a multidoc yaml
+		return nil
```

Comment:
What about docs that have 

```
# service.yaml -- some comment
---
apiVersion: ...
```

Reply:
ah I see its handled here, ok https://github.com/replicatedhq/ship/pull/649/files#diff-5b866f20cba4b13c368860cb9022365fR359

Reply:
should we add a unittest for that case maybe? I think this handles it, but that's a pretty common pattern that has tripped us up in the past.

Reply:
Will do